src/EventGrid/EventGrid/Domain/GetAzureEventGridDomain.cs

```diff
@@ -0,0 +1,190 @@
+
+namespace Microsoft.Azure.Commands.EventGrid
+{
+    /// <summary>
+    /// 'Get-AzureRmEventGridDomain' Cmdlet gives the details of a / List of EventGrid domain(s)
+    /// <para> If Domain name provided, a single Domain details will be returned</para>
+    /// <para> If Domain name not provided, list of Domains will be returned</para>
+    /// </summary>
+
+    [Cmdlet(
+        VerbsCommon.Get,
+        ResourceManager.Common.AzureRMConstants.AzureRMPrefix + "EventGridDomain",
+        DefaultParameterSetName = ResourceGroupNameParameterSet),
+    OutputType(typeof(PSDomain),
+    typeof(PSDomainListInstance))]
+
+    public class GetAzureEventGridDomain : AzureEventGridCmdletBase
+    {
+        [Parameter(
+            Mandatory = true,
+            ValueFromPipelineByPropertyName = true,
+            Position = 0,
+            HelpMessage = EventGridConstants.ResourceGroupNameHelp,
+            ParameterSetName = DomainNameParameterSet)]
+        [Parameter(
+            Mandatory = false,
+            ValueFromPipelineByPropertyName = true,
+            HelpMessage = EventGridConstants.ResourceGroupNameHelp,
+            ParameterSetName = ResourceGroupNameParameterSet)]
+        [ResourceGroupCompleter]
+        [ValidateNotNullOrEmpty]
+        [Alias(AliasResourceGroup)]
+        public string ResourceGroupName { get; set; }
+
+        [Parameter(
+            Mandatory = true,
+            ValueFromPipelineByPropertyName = true,
+            Position = 1,
+            HelpMessage = EventGridConstants.DomainNameHelp,
+            ParameterSetName = DomainNameParameterSet)]
+        [ResourceNameCompleter("Microsoft.EventGrid/domains", nameof(ResourceGroupName))]
+        [ValidateNotNullOrEmpty]
+        [Alias("DomainName")]
+        public string Name { get; set; }
+
+        [Parameter(
+            Mandatory = true,
+            ValueFromPipelineByPropertyName = true,
+            Position = 0,
+            HelpMessage = EventGridConstants.DomainResourceIdHelp,
+            ParameterSetName = ResourceIdEventSubscriptionParameterSet)]
+        [ValidateNotNullOrEmpty]
+        public string ResourceId { get; set; }
+
+        [Parameter(
+            Mandatory = false,
+            ValueFromPipelineByPropertyName = true,
+            HelpMessage = EventGridConstants.ODataQueryHelp,
+            ParameterSetName = DomainNameParameterSet)]
+        [Parameter(
+            Mandatory = false,
+            ValueFromPipelineByPropertyName = true,
+            HelpMessage = EventGridConstants.ODataQueryHelp,
+            ParameterSetName = ResourceGroupNameParameterSet)]
+        [Parameter(
+            Mandatory = false,
+            ValueFromPipelineByPropertyName = true,
+            HelpMessage = EventGridConstants.ODataQueryHelp,
+            ParameterSetName = ResourceIdEventSubscriptionParameterSet)]
+        [ValidateNotNullOrEmpty]
+        public string ODataQuery { get; set; }
```

Comment:
@ahamad-MS do we want to expose this parameter to the user, or introduce new parameters that map to the properties that the user would be providing for `-ODataQuery`?

For example, if the filter supported filtering by some property `Foo`, rather than having the user provide a value to `-ODataQuery` that's `$filter=foo eq 'helloworld'`, should we introduce a `-Foo` parameter that we use to create a filter for the user? We follow this pattern in a few other cmdlets, so I can provide some examples if needed.

Reply:
Thanks for the suggestion but it is not a preferred method for us as the problem is not with parameters themselves rather than the operators. Currently, only parameter/property we filter on is name. However, we have multiple operations with different layout. For example, 1. Contains(name, 'x') 2. name eq x 3. contains(name, 'xyz') or name ne 'z'

We want user to be free to have as complicated query as possible. We added example and enhanced help text for that. Actually, the feedback we got from users so far on having more operations to be supports rather than parameters. Hence, keeping this as is now.